Assets/Scripts/EOSStatsManager.cs

```diff
@@ -0,0 +1,176 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+//#define ENABLE_DEBUG_EOSSTATSMANAGER;
+
+using System.Collections.Generic;
+using Epic.OnlineServices.Stats;
+using System;
+using Epic.OnlineServices;
+using System.Diagnostics;
+
+namespace PlayEveryWare.EpicOnlineServices.Samples
+{
+    /// <summary>
+    /// Class <c>EOSStatsManager</c> is a simplified wrapper for EOS [Stats Interface](https://dev.epicgames.com/docs/services/en-US/Interfaces/Stats/index.html).
+    /// </summary>
+    public class EOSStatsManager : EOSOnConnectLogin, IEOSSubManager
+    {
+        /// <summary>
+        /// Cache of the map between the product user id and the stats for that user.
+        /// </summary>
+        private Dictionary<ProductUserId, List<Stat>> _productUserIdToStatsCache = new();
+
+        protected override void PostSuccessfulLogin()
+        {
+            QueryStatsForProductUserId(_productUserId, (ref OnQueryStatsCompleteCallbackInfo statsQueryData) =>
+            {
+                CacheStatsForProductUserId(_productUserId);
+            });
+        }
+
+        /// <summary>
+        /// Gets the StatsInterface for EOS
+        /// </summary>
+        /// <returns>The stats interface.</returns>
+        private StatsInterface GetEOSStatsInterface()
+        {
+            return EOSManager.Instance.GetEOSPlatformInterface().GetStatsInterface();
+        }
+
+        private void CacheStatsForProductUserId(ProductUserId productUserId)
+        {
+            var statInterface = GetEOSStatsInterface();
+            GetStatCountOptions countOptions = new()
+            {
+                TargetUserId = productUserId
+            };
+            uint statsCountForProductUserId = statInterface.GetStatsCount(ref countOptions);
+
+            List<Stat> collectedStats = new();
+            var copyStatsByIndexOptions = new CopyStatByIndexOptions
+            {
+                TargetUserId = productUserId,
+                StatIndex = 0
+            };
+
+            for (uint i = 0; i < statsCountForProductUserId; ++i)
+            {
+                copyStatsByIndexOptions.StatIndex = i;
+
+                Result copyStatResult = statInterface.CopyStatByIndex(ref copyStatsByIndexOptions, out Stat? stat);
+
+                if (copyStatResult == Result.Success)
+                {
+                    collectedStats.Add(stat.Value);
+                }
+            }
+
+            _productUserIdToStatsCache[productUserId] = collectedStats;
+        }
+
+        private void QueryStatsForProductUserId(ProductUserId productUserId, OnQueryStatsCompleteCallback callback)
+        {
+            if (!productUserId.IsValid())
+            {
+                print("Invalid product user id sent in!");
+                return;
```

Comment:
I feel like we should run the callback with an OnQueryStatsCompleteCallbackInfo that indicates a failure.

Reply:
See [my response](https://github.com/PlayEveryWare/eos_plugin_for_unity/pull/715#discussion_r1635397588) to your other comment.